components/base/sensorcontrolled/spin.go

```diff
@@ -27,11 +29,20 @@ func (sb *sensorBase) Spin(ctx context.Context, angleDeg, degsPerSec float64, ex
 	// If an orientation movement sensor or controls are not configured, we cannot use this Spin method.
 	// Instead we need to use the Spin method of the base that the sensorBase wraps.
 	// If there is no valid velocity sensor, there won't be a controlLoopConfig.
-	if len(sb.controlLoopConfig.Blocks) == 0 {
+	if sb.controlLoopConfig == nil {
 		sb.logger.CWarnf(ctx, "control parameters not configured, using %v's Spin method", sb.controlledBase.Name().ShortName())
 		return sb.controlledBase.Spin(ctx, angleDeg, degsPerSec, extra)
 	}
 
+	// if loop is tuning, return an error
+	// if loop has been tuned but the values haven't been added to the config, error with tuned values
+	if sb.loop != nil && sb.loop.GetTuning(ctx) {
+		return control.TuningInProgressErr(sb.Name().ShortName())
+	} else if (sb.configPIDVals[0].NeedsAutoTuning() && !(*sb.tunedVals)[0].NeedsAutoTuning()) ||
+		(sb.configPIDVals[1].NeedsAutoTuning() && !(*sb.tunedVals)[1].NeedsAutoTuning()) {
+		return control.TunedPIDErr(sb.Name().ShortName(), *sb.tunedVals)
+	}
+
```

Comment:
Ah, is this the same case structure? Then scratch what I said in the above comment. Let's make it a private helper method for the sensor-controlled base.